### PR TITLE
added an overlay based date time picker

### PR DESCRIPTION
**Pull request by BeingBright**
## Checklist before requesting a review
- [ ] Test coverage is adequate
- [ ] Test makes sense
- [ ] Code is documented(all public methods)
- [ ] Code quality is adequate
- [ ] Feature complete
- [ ] (Almost no bugs)
- [ ] Dart analyzer completes without failure
- [ ] Changelog is updated
- [ ] Version is updated
- [ ] Readme is clean

## Why?
The reason why this pull-request is made is because the current date time picker does not work within the project 'Beep'.

### CODE REVIEW

File: lib/src/overlay_date_time_picker.dart [hunk 1]
```diff
@@ -0,0 +1,252 @@
...
+
+  bool _isShown = false;
+
+  late OverlayState? _overlayState;
+  late final OverlayEntry _overlay;
+  late final DateTimePickerController _dateTimePickerController;
+
+  @override
+  void initState() {
+    _overlay = OverlayEntry(
```
Comment (PureTryOut) [inline lib/src/overlay_date_time_picker.dart]: Don't use `initState()` as a constructor when not actually setting state. You can assign the variable immediately when you're declaring it on line 88.

File: lib/src/overlay_date_time_picker.dart [hunk 2]
```diff
@@ -0,0 +1,252 @@
...
+        return Positioned(
+          top: offset.dy,
+          left: offset.dx,
+          child: Material(
+            child: _buildOverlay(context),
+          ),
+        );
+      },
+    );
+    _dateTimePickerController = DateTimePickerController(
```
Comment (PureTryOut) [inline lib/src/overlay_date_time_picker.dart]: Same here, assign it immediately on line 89.

File: lib/src/widgets/overlay_date_time_picker/overlay.dart [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+      usesButtons = true;
+    });
+    _pageController
+        .previousPage(
+      duration: const Duration(milliseconds: 250),
+      curve: Curves.easeInOut,
+    )
+        .then(
+      (value) {
+        previousPage();
```
Comment (PureTryOut) [inline lib/src/widgets/overlay_date_time_picker/overlay.dart]: Why are you splitting this up into another function?

File: lib/src/widgets/overlay_date_time_picker/overlay.dart [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+  void goToPreviousPage() {
+    setState(() {
+      usesButtons = true;
+    });
+    _pageController
+        .previousPage(
+      duration: const Duration(milliseconds: 250),
+      curve: Curves.easeInOut,
+    )
+        .then(
```
Comment (PureTryOut) [inline lib/src/widgets/overlay_date_time_picker/overlay.dart]: Any reason to prefer this over making the entire thing a regular `async` function and awaiting this?

File: lib/src/widgets/overlay_date_time_picker/overlay.dart [hunk 5]
```diff
@@ -0,0 +1,184 @@
...
+  void goToNextPage() {
+    setState(() {
+      usesButtons = true;
+    });
+    _pageController
+        .nextPage(
+      duration: const Duration(milliseconds: 250),
+      curve: Curves.easeInOut,
+    )
+        .then(
```
Comment (PureTryOut) [inline lib/src/widgets/overlay_date_time_picker/overlay.dart]: Any reason to prefer this over making the entire thing a regular `async` function and awaiting this?

File: lib/src/widgets/overlay_date_time_picker/overlay.dart [hunk 6]
```diff
@@ -0,0 +1,184 @@
...
+      (value) {
+        nextPage();
```
Comment (PureTryOut) [inline lib/src/widgets/overlay_date_time_picker/overlay.dart]: Why are you splitting this up into another function?

File: lib/src/widgets/overlay_date_time_picker/pickable_date.dart [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+      case DateBoxShape.circle:
+        return BorderRadius.all(Radius.circular(theme.monthDateBoxSize));
+      case DateBoxShape.roundedRectangle:
+        return const BorderRadius.all(Radius.circular(10));
+      case DateBoxShape.rectangle:
+        return null;
+    }
+  }
+
+  bool isWeekday(DateTime date) {
```
Comment (PureTryOut) [inline lib/src/widgets/overlay_date_time_picker/pickable_date.dart]: This is an unused function. If this is intended as a public API then please add documentation to it.

**Review by PureTryOut — CHANGES_REQUESTED** [review]

**Review by PureTryOut — APPROVED** [review]

**Review by FlutterJoey — APPROVED** [review]

**Pull request merged**